### ISSUE TITLE
dungeon finder? 

### ISSUE THREAD

**Issue by pokeball99** [toolbox4minecraft/amidst#57]
i knowthis was probably asked  a lot over on the mc forum(couldn't find an issue for it on the original repo), but there really should be a option for finding dungeons. while it might be a little much for it to show every possible dungeon( its possible for there to be 8 dungeons to gen in a single chunk), but maybe have it for double or triple or up to a 72 range( the theoretical max number of possible spawners that could be made active at once) 
sorry for the word vomit

**Comment by stefandollase** [comment]
Thanks for the suggestion. You are talking about the dungeons with mossy cobblestone, chests and a zombie, skeleton or spider spawner in the middle, right?

I am relatively new to the project, so maybe @Treer or @BigAlanM know about the complexity of the implementation and why it is not yet included in Amidst. To me, it seems logical to include dungeons.

**Comment by Treer** [comment]
Unfortunately, the rules that determine whether a dungeon is spawned depend on the land being fully generated, i.e. you have to know where all the surrounding blocks will be, and Amidst doesn't generate all the blocks so it can't know where dungeon's will be.

**Comment by Treer** [comment]
We'll probably be asked to add dungeons regularly. For future reference, here's a way for players to find dungeons:

1. First you want to force Minecraft to fully generate each chunk in the area you want to search. It looks like [Minecraft Land Generator](https://sites.google.com/site/minecraftlandgenerator/) will do that for you.

2. Next, you need something to scan through those chunks searching for dungeons. Minutor [can do this](https://www.youtube.com/watch?v=oVfNzA3eGp0), but there might be even better tools out there. Minutor can be found [here](http://seancode.com/minutor).

I've not tried it myself

**Comment by Treer** [comment]
Perhaps someone could add a simple scanner to Amidst that searches for spawners through whatever chunks of the map minecraft has already generated, but this poses a user interface issue because the generated chunks are associated with saved games, whereas Amidst currently displays maps based on profiles.

**Comment by stefandollase** [comment]
Thanks for the clarification. It is really unfortunate that this is not easily possible.

Regarding the user interface issue: Amidst is already able to open specific world saves. This is how the player moving works. Thus, I do not see an issue here.

However, my understanding of this (step 2) is, that we would need to read the region files. We closed issue #19 because we thought reading the region files is out of the scope of Amidst. Do you have a different opinion on that?

**Comment by Treer** [comment]
Good point about moving players = world saves. I'd never used that feature or looked at the code, or looked into the data format. I should probably learn that stuff - it's makes #19 and/or dungeons a much more interesting proposal.

Regarding scope, #19 and dungeons *are* out of scope, but end islands would also have been out of scope. So I reckon if someone went on a coding binge and was able to implement one or more of those features while:
* Smoothly integrating with the existing **simple** user-interface (not over-complicating it).
* Not making a mess of the internal architecture of the program.
* Not adding any brittle or high-maintenance code into the code-base.

Then Amidst would become an even more useful tool. So it's out of scope unless someone wants to gear up and extend the scope themselves. It sounds like a fun challenge and I would definitely be interested if I had more time — but at this stage I don't, plus I have existing features like export to work on)

**Comment by stefandollase** [comment]
Okay, I agree with that opinion. We should find a way to remember feature requests that are good, but currently out of scope.

One way is to just let them open, so they are still visible to everyone.

Another approach is to create a list of them in the wiki entry [Getting Involved](https://github.com/toolbox4minecraft/amidst/wiki/Getting-Involved).

We can also create an extra label for them, called something like **currently out of scope**. This enables everyone to easily find them.

I think I will go with the first and third point and refer to the label from the getting involved page.

**Comment by pokeball99** [comment]
well, http://zipkrowd.com/tools.htm might be of some use, given that its not directly pulled from the mc source code like amidst is now, and that this could also in turn be able to either 1. have a basic plugin system for amidst, or 2. maybe just importing the plugins that have been made so far by those guys(quad witch hut and such)

**Comment by BigAlanM** [comment]
Minutor v2 works quite well for finding dungeons if you download the "dungeon finder" resource pack for it.  
It does this by making all but dungeon specific blocks invisible so you get a black map with only the dungeons and a few scattered blocks showing up.
Of course it can only display dungeons in chunks that have actually been created by running Minecraft with that world and traveling around.

**Comment by pokeball99** [comment]
apon further look into the zipkrowd seed finder, it seems that the guy had made an optimized algorithm for minecraft world generation. its under net/minecraft(1(5/7)) when viewing it in the java decompiler of
so maybe this could be hacked together into amidst or such

**Comment by pokeball99** [comment]
any update on this?

**Comment by stefandollase** [comment]
@pokeball99
This will be out of scope until somebody makes an effort to implement it. I will probably not be the one. This is the reason why it has the currently-out-of-scope label.

**Comment by FreshLine** [comment]
https://chunkbase.com/apps/dungeon-finderchunkbase website is currently doing it so i guess it is possible ?

**Comment by FreshLine** [comment]
> https://chunkbase.com/apps/dungeon-finder chunkbase website is currently doing it so i guess it is possible ?

the chunk generation in their website is slower with dungeon finder than their other structures finding tools so i guess it use and optimized world generation algorithm

**Comment by LDVSOFT** [comment]
ChunkBase dungeon finder looks broken to me, it cannot find spawner I've found in my world right beneath my base.

**Comment by Eiion** [comment]
> ChunkBase dungeon finder looks broken to me, it cannot find spawner I've found in my world right beneath my base.

It works fine for me. But it only finds dungeons not e.g. random spider spawners in mineshafts.
